airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -81,7 +81,9 @@ def _get_conn_params(self) -> dict[str, str | int]:
         conn_params: dict[str, str | int] = {}
 
         if conn.extra_dejson.get("iam", False):
-            conn.login, conn.password, conn.port = self.get_iam_token(conn)
+            conn.login, conn.password, conn.port = self.get_iam_token(
+                conn, is_serverless=conn.extra_dejson.get("is_serverless", False)
+            )
```

Comment:
Since we provide entire Connection object into the `get_iam_token` method, I guess we do not need to use `is_serverless` argument and could resolve it into the `get_iam_token`